projects/packages/forms/src/blocks/field-slider/edit.js

```diff
@@ -32,6 +32,23 @@ export default function SliderFieldEdit( props ) {
 		maxLabel = '',
 	} = attributes;
 
+	const [ localDefault, setLocalDefault ] = useState( String( defaultValue ) );
+	const [ defaultFocused, setDefaultFocused ] = useState( false );
+
+	useEffect( () => {
+		if ( ! defaultFocused ) {
+			setLocalDefault( String( defaultValue ) );
+		}
+	}, [ defaultValue, defaultFocused ] );
+
+	const snapToStep = useCallback(
+		( val, stepSize ) => {
+			const clamped = Math.min( Math.max( val, min ), max );
+			return min + Math.round( ( clamped - min ) / stepSize ) * stepSize;
+		},
+		[ min, max ]
+	);
```

Comment:
Added this snapToStep method, and updated some other areas to use this. There was some duplication since we need to be snapping the step/increment in a couple places, including the new logic for default values.